onadata/libs/utils/cache_tools.py

```diff
@@ -41,3 +44,12 @@ def safe_delete(key):
 def safe_key(key):
     """Return a hashed key."""
     return hashlib.sha256(force_bytes(key)).hexdigest()
+
+
+def cache_single_object(key_prefix, userid, query):
+    key = "{}{}".format(key_prefix, userid)
+    object_list = cache.get(key, None)
+    if object_list is None:
+        object_list, _ = query
+        cache.set(key, object_list)
+    return object_list
```

Comment:
I think this function achieves the purpose of cache. I've noticed we are passing `query` as an arg and using it to set the cache object if doesn't exist. This means we'll always querying the db to retrieve the `query` object then passing it to this function, right? if that's the case, I guess that beats the purpose of caching in the first place, right? We would like to retrieve data from cache first and if doesn't exist from cache, we retrieve the data from database